File: app/src/main/java/com/jz/appframe/di/ViewModelKey.java
package com.jz.appframe.di;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import androidx.lifecycle.ViewModel;
import dagger.MapKey;
import kotlin.annotation.AnnotationRetention;
import kotlin.annotation.MustBeDocumented;

/**
 * @author jackzhous
 * @package com.jz.appframe.di
 * @filename ViewModelKey
 * date on 2020/5/22 7:08 PM
 * @describe TODO
 * @email jackzhouyu@foxmail.com
 **/
@MustBeDocumented
@Target(
        ElementType.METHOD
)
@Retention(RetentionPolicy.RUNTIME)
@MapKey
public @interface ViewModelKey {
    Class<? extends ViewModel> value();
}

File: app/src/main/java/com/jz/appframe/di/ViewModelKey.kt
package com.jz.appframe.di

import androidx.lifecycle.ViewModel
import dagger.MapKey
import kotlin.reflect.KClass

/**
 * @package com.jz.appframe.di
 * @filename ViewModelKey
 * date on 2020/5/27 11:46 AM
 * @author jackzhous
 * @describe TODO
 * @email jackzhouyu@foxmail.com
 **/
@MustBeDocumented
@Target(
        AnnotationTarget.FUNCTION,
        AnnotationTarget.PROPERTY_GETTER,
        AnnotationTarget.PROPERTY_SETTER
)
@Retention(AnnotationRetention.RUNTIME)
@MapKey
annotation class ViewModelKey(val value: KClass<out ViewModel>)